src/SonarQube.Client/Services/SonarQubeClient.cs

```diff
@@ -0,0 +1,284 @@
+                stringResponse => ProcessJsonResponse<PluginResponse[]>(stringResponse, token));
+        }
+
+        public Task<Result<ProjectResponse[]>> GetProjectsAsync(ConnectionRequest connection, CancellationToken token)
+        {
+            const string ProjectsAPI = "api/projects/index"; // Since 2.10
+
+            return InvokeSonarQubeApi(connection, ProjectsAPI, token,
+                stringResponse => ProcessJsonResponse<ProjectResponse[]>(stringResponse, token));
+        }
+
+        public Task<Result<PropertyResponse[]>> GetPropertiesAsync(ConnectionRequest connection, CancellationToken token)
+        {
+            const string PropertiesAPI = "api/properties/"; // Since 2.6
+
+            return InvokeSonarQubeApi(connection, PropertiesAPI, token,
+                stringResponse => ProcessJsonResponse<PropertyResponse[]>(stringResponse, token));
+        }
+
+        public Task<Result<QualityProfileChangeLogResponse>> GetQualityProfileChangeLogAsync(ConnectionRequest connection,
+            QualityProfileChangeLogRequest request, CancellationToken token)
+        {
+            const string QualityProfileChangeLogAPI = "api/qualityprofiles/changelog"; // Since 5.2
+
+            var query = AppendQueryString(QualityProfileChangeLogAPI, "?profileKey={0}&ps={1}",
+                        request.QualityProfileKey, request.PageSize); // TODO: should handle optional params
+
+            return InvokeSonarQubeApi(connection, query, token,
+                stringResponse => ProcessJsonResponse<QualityProfileChangeLogResponse>(stringResponse, token));
+        }
+
+        public Task<Result<QualityProfileResponse[]>> GetQualityProfilesAsync(ConnectionRequest connection,
+           QualityProfileRequest request, CancellationToken token)
+        {
+            const string QualityProfileListAPI = "api/qualityprofiles/search"; // Since 5.2
+
+            var query = request.ProjectKey == null
+                ? AppendQueryString(QualityProfileListAPI, "?defaults=true")
+                : AppendQueryString(QualityProfileListAPI, "?projectKey={0}", request.ProjectKey);
+
+            return InvokeSonarQubeApi(connection, query, token,
+                stringResponse => JObject.Parse(stringResponse)["profiles"].ToObject<QualityProfileResponse[]>());
+        }
+
+        public Task<Result<RoslynExportProfileResponse>> GetRoslynExportProfileAsync(ConnectionRequest connection,
+            RoslynExportProfileRequest request, CancellationToken token)
+        {
+            const string QualityProfileExportAPI = "api/qualityprofiles/export"; // Since 5.2
+
+            var roslynExporterName = string.Format(CultureInfo.InvariantCulture, "roslyn-{0}",
+                        request.LanguageKey);
+            var query = AppendQueryString(QualityProfileExportAPI, "?name={0}&language={1}&exporterKey={2}",
+                request.QualityProfileName, request.LanguageKey, roslynExporterName); // TODO: should handle optional params
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this 'TODO' comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=csharpsquid%3AS1135)